src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,953 @@
+            i += 1
+        else # Check for part ii.
+            isspace(ts[i]) && break
+            ts[i] ∈ ['`', '!', ')', '[', ']', '{', '}', ';', ':', '\'', '"', '.',
+                     ',', '<', '>', '?', '«', '»', '“', '”', '‘', '’'] && continue
+            index_matched = i
+        end
+        i = k
+    end
+
+    index_matched == ts.idx && return false
+    return flushaboutindex!(ts, index_matched)
+end
+
+function nltk_url2(ts)
+    (ts.idx > length(ts.input) || (ts.idx > 1 && ts[ts.idx - 1] == '@')) && return false
+
+    (isascii(ts[ts.idx]) && (isdigit(ts[ts.idx]) || islowercase(ts[ts.idx]))) || return false
+
+    i = ts.idx + 1
+    while i <= length(ts.input) && isascii(ts[i]) && (isdigit(ts[i]) || islowercase(ts[i]))
+        i += 1
+    end
+
+    flush_about = 0
+
+    while i < length(ts.input) && ts[i] ∈ ['.', '-']
+        j = ts[i] == '.' ? i : 0
+        i += 1
+
+        (isascii(ts[i]) && (isdigit(ts[i]) || islowercase(ts[i]))) || break
+        i += 1
+
+        while i <= length(ts.input) && isascii(ts[i]) && (isdigit(ts[i]) || islowercase(ts[i]))
+            i += 1
+        end
+
+        if j != 0 && 3 <= i - j <= 14
+            if i <= length(ts.input) && ts[i] == '/' && (i + 1 > length(ts.input) || ts[i + 1] != '@')
+                flush_about = i
+                break
+            end
+            flush_about = i - 1
+        end
+    end
+
+    flush_about == 0 && return false
+    (flush_about >= length(ts.input) || ts[flush_about + 1] != '@') && return flushaboutindex!(ts, flush_about)
+
+    return false
+end
+
+"""
+    tweet_tokenize(input::AbstractString) => tokens
+
+Twitter-aware tokenizer, designed to be flexible and
+easy to adapt to new domains and tasks.
+
+The basic logic is following:
+
+1. The regular expressions are made for WORD_REGEX (core tokenizer), HANG_REGEX
+   and EMOTICONS_REGEX.
+2  Replacing HTML entities, tweet handles, reducing length of repeated characters
+   and other features, make it suitable for tweets.
+3. The String is tokenized and returned.
+4. `preserve_case` By default is set to `true`. If it is set to `false`,
+   then the tokenizer will downcase everything except for emoticons.
+
+Example:
+
+```
+julia> tweet_tokenize("This is a cooool #dummysmiley: :-) :-P <3 and some arrows < > -> <--")
+
+16-element Array{SubString{String},1}:
+ "This"
+ "is"
+ "a"
+ "cooool"
+ "#dummysmiley"
+ ":"
+ ":-)"
+ ":-P"
+ "<3"
+ "and"
+ "some"
+ "arrows"
+ "<"
+ ">"
+ "->"
+ "<--"
+```
+"""
+function tweet_tokenize(source::AbstractString; strip_handle=false,
+                                            reduce_len=false)
+
+    phonenumbers(ts) = nltk_phonenumbers(ts) || extra_phonenumbers(ts)
+    urls(ts) = nltk_url1(ts) || nltk_url2(ts)
+
+    # Fix HTML Character entities
+    length(source) == 0 && return []
+    source = replace_html_entities(source)
+
+    length(source) == 0 && return []
+    safe_text = pre_process(source, strip_handle, reduce_len)
```

Comment:
`preprocessing` and `handling character entities` could be merged with the main tokenizing function. Doing so should increase speed about 2 to 3 times. I think that I could go for it.

Reply:
Or you can save that for another future PR

Reply:
Sure. I too would prefer that.